clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -222,4 +241,99 @@ private KeyStore load() throws GeneralSecurityException, IOException {
         }
     }
 
+
+    private class ReloadableX509TrustManager extends X509ExtendedTrustManager implements X509TrustManager {
+        private final SecurityStore trustStore;
+        private TrustManagerFactory tmf;
+        private X509TrustManager trustManager;
+        private long lastReload = 0L;
+        private static final long MINIMAL_WAIT = 60 * 1000;
+
+        public ReloadableX509TrustManager(SecurityStore trustStore, TrustManagerFactory tmf) {
+            this.trustStore = trustStore;
+            this.tmf = tmf;
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+            reloadTrustManager();
+            trustManager.checkClientTrusted(chain, authType);
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) throws CertificateException {
+            reloadTrustManager();
+            trustManager.checkServerTrusted(chain, authType);
+        }
+
+        @Override
+        public X509Certificate[] getAcceptedIssuers() {
+            reloadTrustManager();
+            return trustManager.getAcceptedIssuers();
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate[] x509Certificates, String s, Socket socket) throws CertificateException {
+            reloadTrustManager();
+            ((X509ExtendedTrustManager) trustManager).checkClientTrusted(x509Certificates, s, socket);
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] x509Certificates, String s, Socket socket) throws CertificateException {
+            reloadTrustManager();
+            ((X509ExtendedTrustManager) trustManager).checkServerTrusted(x509Certificates, s, socket);
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate[] x509Certificates, String s, SSLEngine sslEngine) throws CertificateException {
+            reloadTrustManager();
+            ((X509ExtendedTrustManager) trustManager).checkClientTrusted(x509Certificates, s, sslEngine);
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate[] x509Certificates, String s, SSLEngine sslEngine) throws CertificateException {
+            reloadTrustManager();
+            ((X509ExtendedTrustManager) trustManager).checkServerTrusted(x509Certificates, s, sslEngine);
+        }
+
+        private void reloadTrustManager() throws KafkaException {
+            try {
+                if (trustManager == null || System.currentTimeMillis() - lastReload > MINIMAL_WAIT) {
```

Comment:
Code has been updated to compare truststore modification time with last reload time. This should address the delay issue and make the result much more predictable. Any chance to get this merged soon? It is time consuming to keep this pull updated with trunk.